### DISCUSSION TITLE
Where can I find verbose logs of PrivateBin?

### DISCUSSION THREAD

**Discussion by tristanlatr** [PrivateBin/PrivateBin#1162] [Q&A]
I'm trying to troubleshoot an issue with my config file not being picked up by PrivateBin, and it's very hard because no error/warnings messages are shown in php or nginx logs :/

Any idea ?

Thanks,

**Comment by elrido** [comment]
The likeliest reason is that the config file is not readable to the php(-fpm) process/user, as non-readable files will be silently ignored:
https://github.com/PrivateBin/PrivateBin/blob/73f8a1d4ae8dc284ceb4ca0b3ccb13bb8534bb5d/lib/Configuration.php#L111-L118
Or it is not in a location that PrivateBin looks at - as you can see above, you can set the environment variable CONFIG_PATH to change the default location that it searches in.

To get more verbose logging you can add some var_dump() calls in the above area against the variables your interested in.

**Comment by tristanlatr** [comment]
Thanks @elrido for your advice, I ended up using this modified version of __construct:

```php
public function __construct()
    {
     	$basePaths  = array();
        $config     = array();
        $configPath = getenv('CONFIG_PATH');
        if ($configPath !== false && !empty($configPath)) {
            $basePaths[] = $configPath;
        }
	else{
             throw new Exception('environment variable CONFIG_PATH is not set', 2);
        }
	$basePaths[] = PATH . 'cfg';
        foreach ($basePaths as $basePath) {
            $configFile = $basePath . DIRECTORY_SEPARATOR . 'conf.php';
            if (is_readable($configFile)) {
                $config = parse_ini_file($configFile, true);
                foreach (array('main', 'model', 'model_options') as $section) {
                    if (!array_key_exists($section, $config)) {
                        throw new Exception(I18n::_('PrivateBin requires configuration section [%s] to be present in configuration file.', $section), 2);
                    }
                }
                break;
            }
            else{
                /* config file not readable! */
                throw new Exception(sprintf('config file %s not readable by user %s', $configFile, get_current_user()), 2);
            }
	}
[...]
```

Which triggered error
`config file /etc/privatebin/conf.php/conf.php not readable by user privatebin`. 

So now I get it, the `CONFIG_PATH` variable should point to the directory containing the `conf.php` file not directly the config file.

**Reply by rugk** [reply]
@elrido Hmm can we maybe do sth. to improve the user experience here? AFAIK this is not the first time we have heard about that mistake. E.g. renaming `CONFIG_PATH`  to `CONFIG_DIRECTORY` or `CONFIG_DIR`. Or just use the provided modified version which shows _some_ exceptions?

**Reply by elrido** [reply]
In hindsight that would have been a better name - I assume whoever implemented it wanted it to be similar sounding to the PATH constant, which also points to a directory.

Unfortunately now we will have to keep supporting it as folks will rely on it in their setups. At most we can add one of the suggested new names as the new default and documented value and retain the older name as deprecated option. Makes it more complex for little gain, IMO.

I'd be fine with logging an error (using `error_log()`), when a file is found but can't be read.

**Reply by rugk** [reply]
> I'd be fine with logging an error (using error_log()), when a file is found but can't be read.'

@tristanlatr as you were at the code already, do you want to take that chance to contribute to PrivateBin (again)?
